### PR TITLE
Coverity scan fixes

### PR DESCRIPTION
**Pull request by lackan**
This PR is in order to fix the coverity detects, totally 13 commits;
First commit is to build the coverity environment, please ignore,
And the other twelve commits is as below:
libtomcrypt(9), libutils(1), libutee(1), core/tee(1)

Maybe some are need to sync up with the mailine, such as libtomcrypt?
Maybe some are not correctly change?
So please comment, and I will take the action, such as pull a request to the libtomcrypt mainline or other.

### CODE REVIEW

File: core/lib/libtomcrypt/src/math/multi.c [hunk 1]
```diff
@@ -59,6 +59,7 @@ int ltc_init_multi(void **a, ...)
               cur = va_arg(clean_list, void**);
           }
           va_end(clean_list);
+          va_end(args);
```
Comment (jenswi-linaro) [inline core/lib/libtomcrypt/src/math/multi.c]: `va_list args` above should be removed instead

File: core/lib/libtomcrypt/src/math/multi.c [hunk 2]
```diff
@@ -87,6 +88,7 @@ int ltc_init_multi_size(int size_bits, void **a, ...)
...
```
Comment (jenswi-linaro) [inline core/lib/libtomcrypt/src/math/multi.c]: `va_list args` above should be removed instead

File: core/lib/libtomcrypt/src/pk/asn1/der/utf8/der_encode_utf8_string.c [hunk 3]
```diff
@@ -64,7 +64,7 @@ int der_encode_utf8_string(const wchar_t *in,  unsigned long inlen,
 
    /* get the size */
    for (x = len = 0; x < inlen; x++) {
-       if (in[x] < 0 || in[x] > 0x1FFFF) {
+       if (in[x] > 0x1FFFF) {
```
Comment (jenswi-linaro) [inline core/lib/libtomcrypt/src/pk/asn1/der/utf8/der_encode_utf8_string.c]: If `wchar_t` is signed or not it implementation defined.

File: core/lib/libtomcrypt/src/pk/asn1/der/utf8/der_length_utf8_string.c [hunk 4]
```diff
@@ -77,7 +77,7 @@ int der_length_utf8_string(const wchar_t *in, unsigned long noctets, unsigned lo
 
    len = 0;
    for (x = 0; x < noctets; x++) {
-      if (in[x] < 0 || in[x] > 0x10FFFF) {
+      if (in[x] > 0x10FFFF) {
```
Comment (jenswi-linaro) [inline core/lib/libtomcrypt/src/pk/asn1/der/utf8/der_length_utf8_string.c]: If `wchar_t` is signed or not it implementation defined.

File: lib/libutee/include/tee_api_defines.h [hunk 5]
```diff
@@ -114,7 +114,7 @@
 #define TEE_DATA_FLAG_SHARE_READ           0x00000010
 #define TEE_DATA_FLAG_SHARE_WRITE          0x00000020
 #define TEE_DATA_FLAG_OVERWRITE            0x00000400
-#define TEE_DATA_MAX_POSITION              0xFFFFFFFF
+#define TEE_DATA_MAX_POSITION              0xFFFFFFFE
```
Comment (jenswi-linaro) [inline lib/libutee/include/tee_api_defines.h]: The standard says `0xFFFFFFFF`

File: lib/libutils/isoc/arch/arm/arm32_aeabi_divmod.c [hunk 6]
```diff
@@ -174,7 +174,7 @@ signed __aeabi_imod(signed numerator, signed denominator)
 	struct qr qr = { .q_n = 0, .r_n = 0 };
 
 	/* in case modulo of a power of 2 */
-	for (i = 0, j = 0, h = 0, s = denominator; (s != 0) || (h > 1); i++) {
+	for (i = 0, j = 0, h = 0, s = denominator; s != 0; i++) {
```
Comment (jenswi-linaro) [inline lib/libutils/isoc/arch/arm/arm32_aeabi_divmod.c]: As I read the code the loop can stop if h > 1 since it's useless after that.

Comment (lackan) [inline lib/libutils/isoc/arch/arm/arm32_aeabi_divmod.c]: I'm not quite understand this, if `h > 1`, the condition `(s != 0) || (h > 1)` will forever be true, and how can it exit the loop?

File: core/lib/libtomcrypt/src/pk/asn1/der/utf8/der_encode_utf8_string.c (hunk 3 above)
Comment (lackan) [inline core/lib/libtomcrypt/src/pk/asn1/der/utf8/der_encode_utf8_string.c]: OK

File: core/lib/libtomcrypt/src/math/multi.c (hunk 1 above)
Comment (lackan) [inline core/lib/libtomcrypt/src/math/multi.c]: Jens, can you please detail it? I can't get your idea, sorry.

Comment (jenswi-linaro) [inline core/lib/libtomcrypt/src/math/multi.c]: Sorry, now that I read the code again I see that your fix is correct.

File: lib/libutils/isoc/arch/arm/arm32_aeabi_divmod.c (hunk 6 above)
Comment (jenswi-linaro) [inline lib/libutils/isoc/arch/arm/arm32_aeabi_divmod.c]: Exactly, the condition should probably be `s && (h <= 1)`
I have a feeling this loop can be optimized/simplified further, but that's not in the scope of this PR

Comment (lackan) [inline lib/libutils/isoc/arch/arm/arm32_aeabi_divmod.c]: Now I understand what you mean, yes, it doesn't case when h > 1, and the logic here should be `s && (h <= 1)`, I will change.

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jbech-linaro** [comment]
@lackan , this has been review by both @jenswi-linaro and @jforissier a week ago. Would you mind rebase again, so we can merge this?

**Comment by lackan** [comment]
@jbech-linaro Sure, I will rebase soon, thanks all.

**Comment by jbech-linaro** [comment]
I'm closing the ticket, either because it has already been answered or that it is no longer relevant or it could be lack of response from the author. Having that said, feel free to re-open the ticket if you have more to add to the ticket.

**Pull request closed without merging**